_server_execute_js.js

```diff
@@ -0,0 +1,257 @@
+          clonedEl.insertBefore(shadowRoot, clonedEl.firstChild);
+        }
+      }
+
+      function replaceShadowRoots(container, clonedContainer) {
+        var elements = container.querySelectorAll('*');
+        var clonedElements = clonedContainer.querySelectorAll('*');
+        [].slice.call(elements).forEach(function(el, i) {
+          replaceShadowRoot(el, clonedElements[i]);
+        });
+        return clonedContainer;
+      }
+
+      function insertShadowStyles(doc, list) {
+        var template = document.createElement('template');
+        template.setAttribute('shadow-styles', '');
+        list.forEach(function(style) {
+          template.content.appendChild(style);
+        });
+        doc.querySelector('head').appendChild(template);
+      }
+
+      function registerShadowRoot() {
+        var t = document.querySelector('template[shadow-styles]');
+        var shadowStyles = t && t.content.children;
+        var proto = Object.create(HTMLElement.prototype);
+        proto.createdCallback = function() {
+          var parent = this.parentNode;
+          if (parent) {
+            var shadowRoot = parent.createShadowRoot();
+            var child;
+            while ((child = this.firstChild)) {
+              if (child.localName == 'shadow-style') {
+                child.remove();
+                child = shadowStyles[child.getAttribute('index')].cloneNode(true);
+              }
+              shadowRoot.appendChild(child);
+            }
+            this.remove();
+          }
+        };
+        document.registerElement('shadow-root', {prototype: proto});
+      }
+
+      function insertPolymerShadowDom() {
+        var head = document.querySelector('head');
+
+        var scriptNode = document.createElement('script');
+        scriptNode.setAttribute('src', 'bower_components/webcomponentsjs/webcomponents-lite.js');
+        head.insertBefore(scriptNode, head.firstChild);
+
+        var script = document.createElement('script');
+        script.textContent = 'window.Polymer = {dom: "shadow", lazyRegister: true}';
+        head.insertBefore(script, head.firstChild.nextSibling);
+      }
+
+      function insertImportLink(hashedURL) {
+        var linkNode = document.createElement('link');
+        linkNode.setAttribute('rel', 'import');
+        linkNode.setAttribute('href', hashedURL);
+        linkNode.setAttribute('async', '');
+        var head = document.querySelector('head');
+        head.appendChild(linkNode);
+      }
+
+      function registerAndReinsert(importURL) {
+        insertPolymerShadowDom();
+        registerShadowRoot();
+        insertImportLink(importURL);
+      }
+
+      function insertScriptsAndImports(clonedDoc, hashedNum) {
+        var scripts = document.createElement('script');
+
+        scripts.textContent = insertPolymerShadowDom.toString();
+        scripts.textContent += String.fromCharCode(13) + registerShadowRoot.toString();
```

Comment:
make another variable for `String.fromCharCode(13)`, no need to recalculate for each use.